src/data-access/repositories/mongo/mongo-shelf-repository.ts

```diff
@@ -38,7 +39,18 @@ export class MongoShelfRepository extends BaseMongoRepository implements ShelfCo
 
   public async updateShelfConfiguration(shelfConfiguration: ShelfConfiguration): Promise<ShelfConfiguration> {
     this.assertDatabaseConnection(this.updateShelfConfiguration.name)
+    shelfConfiguration = this.applyMissingActionPanelIds(shelfConfiguration)
     await this.getCollection().updateOne({ id: shelfConfiguration.id }, { $set: shelfConfiguration })
     return shelfConfiguration
   }
+
+  private applyMissingActionPanelIds(shelfConfiguration: ShelfConfiguration): ShelfConfiguration {
+    shelfConfiguration.actionPanelConfigurations = shelfConfiguration.actionPanelConfigurations.map(actionPanel => {
+      if (!actionPanel.id || actionPanel.id.length <= 0) {
+        actionPanel.id = this.uuidGenerator.generateUuid()
+      }
+      return actionPanel
+    })
```

Comment:
Condensing the condition to `!actionPanel.id` should give the same behavior since an empty string is considered a falsey value.

If you want to use a bit of fancy syntax you can condense this down to:
```typescript
    shelfConfiguration.actionPanelConfigurations = shelfConfiguration.actionPanelConfigurations.map(actionPanel => {
      actionPanel.id ||= this.uuidGenerator.generateUuid()
      return actionPanel
    })
```
or via a functional approach:
```typescript
    shelfConfiguration.actionPanelConfigurations = shelfConfiguration.actionPanelConfigurations.map(actionPanel => ({
      ...actionPanel,
      id: actionPanel.id || this.uuidGenerator.generateUuid()
    }))
```
or have the entire method use functional composition, like the below snippet. I have changed `actionPanel` to `actionPanelConfiguration`.
```typescript
    return {
      ..shelfConfiguration,
      actionPanelConfigurations: shelfConfiguration.actionPanelConfigurations.map(
        actionPanelConfiguration => ({
          ...actionPanelConfiguration,
          id: actionPanel.id || this.uuidGenerator.generateUuid()
        })
      )
    }
```

Reply:
I have changed the conditional to just be `!actionPanel.id`.
I like the simplicity of that approach compared to some of your functional-approach suggestions.